tests/hw-accel/nfd/internal/wait/wait.go

```diff
@@ -29,30 +36,49 @@ func ForLabel(apiClient *clients.Settings, timeout time.Duration, label string)
 				return false, e
 			}
 
-			labelExist := false
 			workernodes := filterNodesByLabel(nodes, "worker")
+			if len(workernodes) == 0 {
+				glog.V(nfdparams.LogLevel).Infof("No worker nodes found")
+
+				return false, nil
+			}
+
+			nodesWithLabels := 0
+			totalWorkerNodes := len(workernodes)
 
 			for _, node := range workernodes {
-				labelKeys := getLabelKeys(node.Labels)
-				onelineLabels := strings.Join(labelKeys, ", ")
+				featureLabelCount, sampleLabels := countFeatureLabels(node.Labels, label)
+				hasFeatureLabels := featureLabelCount > 0
 
-				if strings.Contains(onelineLabels, label) {
-					labelExist = true
-				} else {
-					labelExist = false
+				if hasFeatureLabels {
+					nodesWithLabels++
 
-					break
+					glog.V(nfdparams.LogLevel).Infof("Node %s has %d feature labels (examples: %v)",
+						node.Name, featureLabelCount, sampleLabels)
+				} else {
+					glog.V(nfdparams.LogLevel).Infof("Node %s does not have feature labels yet", node.Name)
 				}
 			}
 
-			if labelExist {
+			glog.V(nfdparams.LogLevel).
+				Infof("Feature label progress: %d/%d worker nodes have labels",
+					nodesWithLabels,
+					totalWorkerNodes)
+
+			allNodesLabeled := (nodesWithLabels == totalWorkerNodes)
```

Comment:
nit: imho a redundant variable (you use it only once). Just do the comparison in the if statement on line 70